addons/netfox/rollback/rollback-synchronizer.gd

```diff
@@ -148,27 +151,63 @@ func _process_tick(tick: int):
 
 func _record_tick(tick: int):
 	# Broadcast state we own
-	if not _auth_state_props.is_empty():
-		var broadcast = {}
+	if not _auth_state_props.is_empty(): #This is always the server, as the server owns all avatars
+		var full_state_to_broadcast: Dictionary = {}
+		if (multiplayer.is_server()):
+			_logger.info("recorded tick %s for %s" % [tick, root.player_id])
 
 		for property in _auth_state_props:
 			if _can_simulate(property.node, tick - 1):
 				# Only broadcast if we've simulated the node
-				broadcast[property.to_string()] = property.get_value()
-	
-		if broadcast.size() > 0:
+				full_state_to_broadcast[property.to_string()] = property.get_value()
+
+		if full_state_to_broadcast.size() > 0:
 			# Broadcast as new state
-			_latest_state = max(_latest_state, tick)
-			_states[tick] = PropertySnapshot.merge(_states.get(tick, {}), broadcast)
-			_submit_state.rpc(broadcast, tick)
-	
+			_latest_state_tick = max(_latest_state_tick, tick)
+			_states[tick] = PropertySnapshot.merge(_states.get(tick, {}), full_state_to_broadcast)
+
+			if (NetworkRollback.enable_state_diffs == false || _states.has(tick - 1) == false):
+				_submit_state.rpc(full_state_to_broadcast, tick)
+			else:
+				var properties_to_include: Array[String] = []
+				var diff_state_to_broadcast: Dictionary = {}
+				for picked_property_path in full_state_to_broadcast:
+					#If previous tick doesnt have a property, this means its a new property added in runtime, so we must add it.
+					if (_states[tick - 1].has(picked_property_path) == false):
+						properties_to_include.append(picked_property_path)
+					#If different value, include it in broadcasting state
+					elif (_states[tick - 1][picked_property_path] != full_state_to_broadcast[picked_property_path]):
+						properties_to_include.append(picked_property_path)
+
+				#Set property values which are different from previous state sent
+				for picked_property in properties_to_include:
+					diff_state_to_broadcast[picked_property] = full_state_to_broadcast[picked_property]
+
+				for picked_peer_id in multiplayer.get_peers():
+					if (NetworkTime._synced_clients.has(picked_peer_id) == false):
```

Comment:
Ah right, and if the state jumps from 54 to 60, we can't do diff states because we don't have tick 59, right? If so, I'd refer to my [other comment] on the approach. Ideally we shouldn't need a contiguous state history. 

[other comment]: https://github.com/foxssake/netfox/pull/278#issuecomment-2415184243